File: Contests/2019-2020/Bronze/Jan/word.cpp
#include <bits/stdc++.h>
#pragma GCC optimize(2)
using namespace std;
typedef long long ll;
const int MAXN = 105;

int n, k, cnt;
string s;
int main(){
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    freopen("word.in", "r", stdin);
    freopen("word.out", "w", stdout);
    cin >> n >> k >> s;
    cout << s;
    cnt = s.size();
    while (--n) {
        cin >> s;
        if (cnt + s.size() <= k) {
            cout << " " << s;
            cnt += s.size();
        } else {
            cout << "\n" << s;
            cnt = s.size();
        }
    }
    return 0;
}

File: Contests/2019-2020/Bronze/Jan/word.py
import sys
sys.stdin = open("word.in", "r")
sys.stdout = open("word.out", "w")

n, k = map(int, input().split())
a = input().split()
print(a[0], end = '')
cnt = len(a[0])
for i in range(1, n):
    if cnt + len(a[i]) <= k:
        print(' ' + a[i], end = '')
        cnt += len(a[i])
    else:
        print('\n' + a[i], end = '')
        cnt = len(a[i])